compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -456,26 +456,32 @@ impl<'a> Resolver<'a> {
                 // the further the two are apart, the higher the chance of the suggestion being wrong
                 // also make sure that the pos for the suggestion is not 0 (ICE #90878)
 
-                let sp =
-                    self.session.source_map().span_extend_to_prev_str(ident.span, current, true);
+                let sp = self
+                    .session
+                    .source_map()
+                    .span_extend_to_prev_str(ident.span, current, true, false);
 
-                let pos_for_suggestion = sp.lo().0.saturating_sub(current.len() as u32);
+                let pos_for_suggestion =
+                    sp.map(|sp| sp.lo().0.saturating_sub(current.len() as u32));
```

Comment:
This saturating_sub and the check for `pos_for_suggestion == 0` were needed to avoid an overflow when the pattern was not found. Now that should be impossible because in that case `span_extend_to_prev_str` returns None. So you could just remove this variable and inline it later:

```rust
let sp = sp.with_lo(BytePos(sp.lo().0 - (current.len() as u32)));
```

Also there's another outdated comment above this line, 

`also make sure that the pos for the suggestion is not 0`

that case is now `None` so the check is implicit, you could mention "only calculate suggestion if the span is found" but that's redundant, so just remove that line.

Reply:
See the above comment